docs/usage/configfile.md

```diff
@@ -71,7 +71,7 @@ volumes: # repeatable flags are represented as YAML lists
 ports:
   - port: 8080:80 # same as `--port '8080:80@loadbalancer'`
     nodeFilters:
-      - loadbalancer
+      - loadbalancer # can be one of: server/agent/loadbalancer/all
```

Comment:
Note that server and agent need to have an index, e.g. `agent[0]` or `server[*]` and it will fail without this.
Maybe you could write those options as (commented out) list items?
A small section in the `Internals/` part of the docs may be a good idea as well to inform about what node filters are and how they can be used :thinking:

Reply:
Yes definitely the index should go there, i just tried to keep the line short, as there are agent and server examples a couple of line below. But would be better to include it.

Hmm I could be even go into `usage/commands.md` or actually included in the cli --help output?

I just realised Nodefilter is related to: envs/labels/volumes/ports. So maybe it deserves its own page, and reference it from multiple place?


Right now the --help display this:
```
  -p, --port [HOST:][HOSTPORT:]CONTAINERPORT[/PROTOCOL][@NODEFILTER]   Map ports from the node containers to the host (Format: [HOST:][HOSTPORT:]CONTAINERPORT[/PROTOCOL][@NODEFILTER])
                                                                        - Example: `k3d cluster create --agents 2 -p 8080:80@agent[0] -p 8081@agent[1]`
```
Where you get some useful examples, but not the possible values/rules.

Which one would you prefer:
- puting a short description about nodefilters right into the --help output. Maybe like a few other examples covering all values and all index syntax
- or just put a link do the docs site, where you have place for more detailed explanation

Reply:
I think I would create a page `nodefilters` under [`docs/internals/`](https://github.com/rancher/k3d/tree/main/docs/internals) that lists all possible values with a small explanation, which we can then link to from other docs and the help text.
Adding it to the help text would mean editing several different parts in the code everytime that something changes to the concept of nodefilters.